### PR TITLE
Add Go code for linked_list

### PR DESCRIPTION
**Pull request by dlfld**
完成linked_list部分Go语言代码

### CODE REVIEW

File: codes/go/chapter_array_and_linkedlist/linked_list_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    n3 := &ListNode{4,nil}
+    n4 := &ListNode{5,nil}
+    // 构建引用指向
+    n0.Next = n1
+    n1.Next = n2
+    n2.Next = n3
+    n3.Next = n4
+
+    for n0 != nil{
+        fmt.Printf("%v ",n0.Val)
```
Comment (krahets) [inline codes/go/chapter_array_and_linkedlist/linked_list_test.go:27]: Use `PrintLinkedList()` function in `print_utils.go`

**Review by krahets — COMMENTED** [review]
Thanks! Please address the comments.

**Comment by dlfld** [comment]
是这样吗？

File: codes/go/chapter_array_and_linkedlist/linked_list_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    n2 := &ListNode{3,nil}
+    n3 := &ListNode{4,nil}
+    n4 := &ListNode{5,nil}
+    // 构建引用指向
+    n0.Next = n1
+    n1.Next = n2
+    n2.Next = n3
+    n3.Next = n4
+
+    PrintLinkedList(n0)
```
Comment (krahets) [inline codes/go/chapter_array_and_linkedlist/linked_list_test.go:23]: Where is the code to test `linked_list.go`?

File: codes/go/chapter_array_and_linkedlist/linked_list.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
+// File: linked_list.go
+// Created Time: 2022-12-23
+// Author: dlfld (2441086385@qq.com)
+
+package chapter_array_and_linkedlist
+/* 链表结点结构体 */
+type ListNode struct {
+    Val int // 节点值
+    Next *ListNode // 指向下一结点的指针（引用）
```
Comment (krahets) [inline codes/go/chapter_array_and_linkedlist/linked_list.go:9]: gofmt is required to format the code.

**Review by krahets — COMMENTED** [review]
Several comments to be addressed

**Comment by krahets** [comment]
Closed due to no response from the author.

**Pull request closed without merging**